v8env/ts/body_mixin.ts

```diff
@@ -93,7 +93,19 @@ export default class BodyMixin implements Body {
   }
 
   async arrayBuffer(): Promise<ArrayBuffer> {
-    if (this.bodySource instanceof ArrayBuffer) {
+    if (this.bodySource instanceof Int8Array ||
+        this.bodySource instanceof Int16Array ||
+        this.bodySource instanceof Int32Array ||
+        this.bodySource instanceof Uint8Array ||
+        this.bodySource instanceof Uint16Array ||
+        this.bodySource instanceof Uint32Array ||
+        this.bodySource instanceof Uint8ClampedArray ||
+        this.bodySource instanceof Float32Array ||
+        this.bodySource instanceof Float64Array
+    ) {
+      const buffer = this.bodySource.buffer
+      if (buffer instanceof ArrayBuffer) return buffer
```

Comment:
Isn't that always true? If it's not always true, it should probably throw an error or else the body will be `undefined` and that would be unexpected.

Reply:
It is not always true, it could also be a `SharedArrayBuffer`. And if nothing returns it makes its way to the [bottom](https://github.com/superfly/fly/blob/master/v8env/ts/body_mixin.ts#L112-L113) where the appropriate error is thrown.